### DISCUSSION TITLE
Update Not Working

### DISCUSSION THREAD

**Discussion by gcortes** [community-scripts/ProxmoxVE#239] [Q&A]
I haven't done an update in a while so I may be doing some wrong. I typed the command in the console for node-red. It says the command is not recognized. The scripts website says that's how you do it and I'm pretty I've run it in the past. It also says I could this: 
bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/node-red.sh)"

I haven't tried it because I want to be sure it doesn't do a complete reinstall and wipe out my data.

I also tried update on Home Assistant. It ran, but died with an error. I know it used to work.  I did set a post about a similar error in another script.

**Comment by MickLesk** [comment]
Can you share more information?

What say the whiptail dialogue? Can you share some screenshots?

**Comment by Wolffoxfangs** [comment]
Did you run the Proxmox Update Repositories script in your proxmox shell first before you tried to update? Might be trying to reference old repos that arent able to be maintained instead of the community repos?

**Comment by gcortes** [comment]
I should have brought up one problem at a time. If you're asking about Home Assistant, I got the following error:
Updating to Stable Version
 ✓ Stopped Home Assistant
 / Updating Home Assistant   
[ERROR] in line 88: exit code 0: while executing command uv pip install ${BR}--upgrade homeassistant &> /dev/null

root@homeassistant-core:~# 

I've run update a number of times on this instance without issue. 
I've never run the Proxmox Update Repositories script. I didn't know there was one.
I don't know or have forgotten what the whiptail dialogue is.

## Update
I ran the update repositories script. It gave a few errors on some of my LXC's, but not Home Assistant. I reran the update afterwards and got the same error.

**Reply by MickLesk** [reply]
Which Script do you use?

**Reply by gcortes** [reply]
Are you asking what script I used to do the update? If so, I just typed update on the console prompt.

**Reply by MickLesk** [reply]
Can you share a Screenshot of this Output:

nano /usr/bin/update

**Reply by gcortes** [reply]
Just one line:
bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/homeassistant-core.sh)"

When I run the script, I take all the defaults.

**Reply by MickLesk** [reply]
okay i just wanted to make sure it wasn't still referencing the old repo.

**Comment by knario** [comment]
Hi, yesterday I noticed that if I do in some lxc. apt update && apt upgrade -y and then run the update command to update using the script, that's when it gives problems.
If I run update first and then run apt update && apt upgrade -y , then it doesn't give problems when updating the lxc.
In case it helps anyone.
What I haven't tested is if doing it this way, it will work for future updates.
Best regards.

**Reply by MickLesk** [reply]
What? :smile:

Can you declare a little bit better?

**Reply by knario** [reply]
hahahahaha !!! sorry.
I use deepl translator to be able to write here in English. 
Yesterday I was updating the debian or ubuntu of the lxc's I have installed with apt update + apt upgrade. After updating each debian or ubuntu, I noticed that there was an update with a new version for Paperless_ngx. 
I had already updated this container with apt update and upgrade. Then I ran the update command to jump to the new version of paperless_ngx and got an error.
I did this same process on several lxc's that had new versions available and the same error.
So, my solution was to recover a backup from the previous day and first update and then upgrade debian or ubuntu on each lxc.

I don't know if you understand me now, sorry.

**Comment by te0203** [comment]
I also have an error after the update node-red. I installed a new container and the error immediately.

Stopped nodered.service - Node-RED.
nodered.service: Start request repeated too quickly.
nodered.service: Failed with result 'signal'.
Failed to start nodered.service - Node-RED.

Any ideas on how to deal with this?

**Reply by MickLesk** [reply]
do an issue. Not on closed discussions. 
Post Screenshots and run in verbose mode

**Reply by gcortes** [reply]
I can do it. How do you do verbose mode? Is it an option when the script runs? I don't remember seeing it.

**Reply by te0203** [reply]
Opened a topic. https://github.com/community-scripts/ProxmoxVE/discussions/624
This is when the script starts.
It seems to be a problem on the server


I installed the container again. During installation, a line popped up

\update failed - see /var/log/pveam.log for details

/var/log/pveam.log

2024-12-01 22:28:28 starting update
2024-12-01 22:28:28 start download http://download.proxmox.com/images/aplinfo-pve-8.dat.asc
2024-12-01 22:28:30 download finished: 200 OK
2024-12-01 22:28:30 start download http://download.proxmox.com/images/aplinfo-pve-8.dat.gz
2024-12-01 22:28:30 download finished: 200 OK
2024-12-01 22:28:30 signature verification: gpgv: Signature made Tue Nov 19 15:13:43 2024 EET
2024-12-01 22:28:30 signature verification: gpgv: using RSA key F4E136C67CDCE41AE6DE6FC81140AF8F639E0C39
2024-12-01 22:28:30 signature verification: gpgv: Good signature from "Proxmox Bookworm Release Key [proxmox-release@proxmox.com](mailto:proxmox-release@proxmox.com)"
2024-12-01 22:28:30 update successful
2024-12-01 22:28:30 start download https://releases.turnkeylinux.org/pve/aplinfo.dat.asc
2024-12-01 22:28:30 download failed: 521 Unknown code
2024-12-01 22:28:30 update failed - no signature file '/var/lib/pve-manager/apl-info/pveam-releases.turnkeylinux.org.tmp.3398047.asc'

**Reply by MickLesk** [reply]
> I can do it. How do you do verbose mode? Is it an option when the script runs? I don't remember seeing it.

Advanced Mode, enter all and in the Last or pre Last Section Switch to yes If verbose

**Reply by gcortes** [reply]
Sorry to be so clueless. How do you enter advanced mode?

I found an example. It shows it as step in the script. I don't see that step.

**Reply by MickLesk** [reply]
![image](https://github.com/user-attachments/assets/b7ba8556-1d38-4c58-8fd2-950bd7935977)

**Comment by te0203** [comment]
Clean installation. No errors during installation. Same result.

root@node-red:~# node-red-log
Started nodered.service - Node-RED.
nodered.service: Main process exited, code=killed, status=4/ILL
nodered.service: Failed with result 'signal'.
nodered.service: Scheduled restart job, restart counter is at 1.
Stopped nodered.service - Node-RED.
...........
Started nodered.service - Node-RED.
nodered.service: Main process exited, code=killed, status=4/ILL
nodered.service: Failed with result 'signal'.
nodered.service: Scheduled restart job, restart counter is at 7.
Stopped nodered.service - Node-RED.
nodered.service: Start request repeated too quickly.
nodered.service: Failed with result 'signal'.
Failed to start nodered.service - Node-RED.

Tried to change
/usr/lib/node_modules/node-red/ewttings.js logging: console: level: "debug"
Nothing has changed

**Comment by te0203** [comment]
Full installation log.
[log.txt](https://github.com/user-attachments/files/17974572/log.txt)

root@node-red:~# node-red-log
/etc/systemd/system/nodered.service:11: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, and consider removing the setting altogether.
Started nodered.service - Node-RED.
nodered.service: Main process exited, code=killed, status=4/ILL
nodered.service: Failed with result 'signal'.
nodered.service: Scheduled restart job, restart counter is at 1.
Stopped nodered.service - Node-RED.
...........
Started nodered.service - Node-RED.
nodered.service: Main process exited, code=killed, status=4/ILL
nodered.service: Failed with result 'signal'.
nodered.service: Scheduled restart job, restart counter is at 5.
Stopped nodered.service - Node-RED.
nodered.service: Start request repeated too quickly.
nodered.service: Failed with result 'signal'.
Failed to start nodered.service - Node-RED.
